File: examples/effectsize/example_cohensd.R
library("effectsize")
data(Boston, package="MASS")
# one sample
index  <- sample(nrow(Boston), size=100)
d      <- cohens_d(Boston$medv[index], mu=10)
d
interpret_cohens_d(d)
# two samples
xind   <- index[Boston$chas[index]==0]
yind   <- index[Boston$chas[index]==1]
d      <- cohens_d(x=Boston$medv[xind], y=Boston$medv[yind])
d
interpret_cohens_d(d)

File: examples/effectsize/example_cohensd.py
import pandas as pd, numpy as np, pingouin as pg
from scipy.stats import ttest_1samp
Boston = r["MASS::Boston"]
# One sample not available
# Two samples
x = Boston.loc[Boston['chas'] == 0, 'medv']
y = Boston.loc[Boston['chas'] == 1, 'medv']
d_two_samples = pg.compute_effsize(x, y, eftype='cohen')
print('Cohen\'s d (two samples):', d_two_samples)
